core/lib/instrument.c

```diff
@@ -2974,7 +2974,8 @@ raw_mem_alloc(size_t size, uint prot, void *addr, dr_alloc_flags_t flags)
             : (TEST(DR_ALLOC_COMMIT_ONLY, flags) ? RAW_ALLOC_COMMIT_ONLY : 0);
 #    endif
         if (IF_WINDOWS(TEST(DR_ALLOC_COMMIT_ONLY, flags) &&) addr != NULL &&
-            !app_memory_pre_alloc(get_thread_private_dcontext(), addr, size, prot, false))
+            !app_memory_pre_alloc(get_thread_private_dcontext(), addr, size, prot, false,
+                                  true /*update*/, false /*!image*/))
```

Comment:
Maybe {}, even though no multi-line body?

Reply:
Not sure what the reasoning is, but I can add it.  Done.